chainer/datasets/text_dataset.py

```diff
@@ -0,0 +1,172 @@
+            Name of the encoding used to decode the file.
+            See the description in :func:`open` for the supported options and
+            how it works.
+            When reading from multiple text files, you can also pass a list of
+            :class:`str` to use different encoding for each file.
+        errors (str or list of str):
+            String that specifies how decoding errors are to be handled.
+            See the description in :func:`open` for the supported options and
+            how it works.
+            When reading from multiple text files, you can also pass a list of
+            :class:`str` to use different error handling policy for each file.
+        newline (str or list of str):
+            Controls how universal newlines mode works.
+            See the description in :func:`open` for the supported options and
+            how it works.
+            When reading from multiple text files, you can also pass a list of
+            :class:`str` to use different mode for each file.
+        filter_func (callable):
+            Function to filter each line of the text file.
+            It should be a function that takes number of arguments equals to
+            the number of files. Arguments are lines loaded from each file.
+            The filter function must return True to accept the line, or
+            return False to skip the line.
+
+    """
+
+    _fps = None
+
+    def __init__(
+            self, paths, encoding=None, errors=None, newline=None,
+            filter_func=None):
+        if isinstance(paths, six.string_types):
+            paths = [paths]
+        elif len(paths) == 0:
+            raise ValueError('at least one text file must be specified')
+
+        if isinstance(encoding, six.string_types) or encoding is None:
+            encoding = [encoding] * len(paths)
+        if isinstance(errors, six.string_types) or errors is None:
+            errors = [errors] * len(paths)
+        if isinstance(newline, six.string_types) or newline is None:
+            newline = [newline] * len(paths)
+
+        if not (len(paths) == len(encoding) == len(errors) == len(newline)):
+            raise ValueError(
+                'length of each option must match with the number of '
+                'text files to read')
+
+        self._paths = paths
+        self._encoding = encoding
+        self._errors = errors
+        self._newline = newline
+        self._fps = None
+
+        self._open()
+
+        # Line number is 0-origin.
+        # `lines` is a list of line numbers not filtered; if no filter_func is
+        # given, it is range(linenum)).
+        # `bounds` is a list of cursor positions of line boundaries for each
+        # file, i.e. i-th line of k-th file starts at `bounds[k][i]`.
+        linenum = 0
+        lines = []
+        bounds = tuple([[0] for _ in self._fps])
+        while True:
+            data = [fp.readline() for fp in self._fps]
```

Comment:
What happens when there is an empty line?

Reply:
Empty lines will become `\n` (or `\r` or `\r\n`).

Reply:
Added test case for this.